src/udiskslinuxblock.c

```diff
@@ -487,97 +489,138 @@ process_device_symlinks (const gchar      *device,
+      if (symlinks && g_strv_contains (symlinks, device))
+        return TRUE;
+    }
+  if (label != NULL && g_strcmp0 (label, udisks_block_get_id_label (UDISKS_BLOCK (block))) == 0)
+    {
+      return TRUE;
+    }
+  if (uuid != NULL && g_strcmp0 (uuid, udisks_block_get_id_uuid (UDISKS_BLOCK (block))) == 0)
+    {
+      return TRUE;
+    }
+  if (partlabel != NULL || partuuid != NULL)
+    {
+      UDisksLinuxBlockObject *object;
+      UDisksLinuxDevice *linux_device;
 
-      if (device != NULL)
-        {
-          process_device_symlinks (device, block, entry, &ret);
-        }
-      else if (label != NULL && g_strcmp0 (label, udisks_block_get_id_label (UDISKS_BLOCK (block))) == 0)
-        {
-          ret = g_list_prepend (ret, g_object_ref (entry));
-        }
-      else if (uuid != NULL && g_strcmp0 (uuid, udisks_block_get_id_uuid (UDISKS_BLOCK (block))) == 0)
-        {
-          ret = g_list_prepend (ret, g_object_ref (entry));
-        }
-      else if (partlabel != NULL || partuuid != NULL)
+      object = udisks_daemon_util_dup_object (block, NULL);
+      if (object != NULL)
         {
-          UDisksLinuxBlockObject *object;
-          UDisksLinuxDevice *linux_device;
-
-          object = udisks_daemon_util_dup_object (block, NULL);
-          if (object == NULL)
-            goto continue_loop;
           linux_device = udisks_linux_block_object_get_device (object);
           g_clear_object (&object);
-          if (linux_device == NULL)
-            goto continue_loop;
-          if (linux_device->udev_device == NULL)
+          if (linux_device != NULL && linux_device->udev_device != NULL &&
+              ((partuuid != NULL  && g_strcmp0 (partuuid,  g_udev_device_get_property (linux_device->udev_device, "ID_PART_ENTRY_UUID")) == 0) ||
+               (partlabel != NULL && g_strcmp0 (partlabel, g_udev_device_get_property (linux_device->udev_device, "ID_PART_ENTRY_NAME")) == 0)))
             {
               g_object_unref (linux_device);
-              goto continue_loop;
+              return TRUE;
             }
-          if ((partuuid != NULL && g_strcmp0 (partuuid, g_udev_device_get_property (linux_device->udev_device, "ID_PART_ENTRY_UUID")) == 0) ||
-              (partlabel != NULL && g_strcmp0 (partlabel, g_udev_device_get_property (linux_device->udev_device, "ID_PART_ENTRY_NAME")) == 0))
-            ret = g_list_prepend (ret, g_object_ref (entry));
-          g_object_unref (linux_device);
+          g_clear_object (&linux_device);
         }
+    }
 
-    continue_loop:
-      ;
+  return FALSE;
+}
+
+static GList *
+find_fstab_entries (UDisksDaemon     *daemon,
+                    UDisksLinuxBlock *block,
+                    const gchar      *needle)
+{
+  struct libmnt_table *table;
+  struct libmnt_iter* iter;
+  struct libmnt_fs *fs = NULL;
+  GList *ret = NULL;
+
+  table = mnt_new_table ();
+  if (mnt_table_parse_fstab (table, NULL) < 0)
+    {
+      mnt_unref_table (table);
```

Comment:
`mnt_unref_table` is not available in libmount < 2.24 so we probably should use `mnt_free_table` instead

Reply:
Gah, `libmount` could really use `@since` annotations. So much for the `Don't use this function directly` note in the docs.